### PR TITLE
jsonnet: move resource requests assignment from Pods to Containers

### PR DESCRIPTION
**Pull request by paulfantom**
In #356 I have assigned resource requests to pod object instead of containers. This should fix that mistake.

Note: `Prometheus` object should implicitly pass resource requests to prometheus container

/cc: @metalmatze @s-urbaniak

### CODE REVIEW

File: jsonnet/kube-state-metrics.jsonnet [hunk 1]
```diff
@@ -80,12 +80,24 @@ local tlsVolumeName = 'kube-state-metrics-tls';
                         volumeMounts: [
                           containerVolumeMount.new(tlsVolumeName, '/etc/tls/private'),
                         ],
-                        resources: {},
+                        resources: {
+                          requests: {
+                            memory: '120Mi',
```
Comment (brancz) [inline jsonnet/kube-state-metrics.jsonnet]: data shows 40Mi

File: jsonnet/prometheus-operator.jsonnet [hunk 2]
```diff
@@ -7,18 +7,17 @@
...
-              },
               containers:
                 std.map(
                   function(c) c {
-                    resources: {},
                     args+: ['--namespaces=' + $._config.namespace],
                     securityContext: {},
+                    resources: {
+                      requests: {
+                        memory: '100Mi',
```
Comment (brancz) [inline jsonnet/prometheus-operator.jsonnet]: our data showed 60Mi to be appropriate

File: jsonnet/prometheus-adapter.jsonnet [hunk 3]
```diff
@@ -74,6 +74,12 @@ local tlsVolumeName = 'kube-state-metrics-tls';
                           containerVolumeMount.new(prometheusAdapterPrometheusConfig, prometheusAdapterPrometheusConfigPath),
                           containerVolumeMount.new(servingCertsCABundle, servingCertsCABundleMountPath),
                         ],
+                        resources: {
+                          requests: {
+                            memory: '50Mi',
```
Comment (brancz) [inline jsonnet/prometheus-adapter.jsonnet]: data shows 20Mi

File: jsonnet/kube-state-metrics.jsonnet [hunk 4]
```diff
@@ -80,12 +80,24 @@ local tlsVolumeName = 'kube-state-metrics-tls';
...
+                        },
                       }
                     else
                       c +
                       container.withVolumeMounts([containerVolumeMount.new(tmpVolumeName, '/tmp')]) +
-                      { resources: {} },
+                      { 
+                        resources: {
+                          requests: {
+                            memory: '120Mi',
```
Comment (brancz) [inline jsonnet/kube-state-metrics.jsonnet]: data shows 20Mi

**Comment by paulfantom** [comment]
Changed memory requests to suggested values

**Pull request merged**